codegen-client/src/test/kotlin/software/amazon/smithy/rust/codegen/client/smithy/protocols/eventstream/ClientEventStreamUnmarshallerGeneratorTest.kt

```diff
@@ -7,43 +7,59 @@ package software.amazon.smithy.rust.codegen.client.smithy.protocols.eventstream
 
 import org.junit.jupiter.params.ParameterizedTest
 import org.junit.jupiter.params.provider.ArgumentsSource
-import software.amazon.smithy.codegen.core.Symbol
-import software.amazon.smithy.model.shapes.StructureShape
-import software.amazon.smithy.rust.codegen.client.smithy.ClientCodegenContext
-import software.amazon.smithy.rust.codegen.core.smithy.CodegenTarget
-import software.amazon.smithy.rust.codegen.core.smithy.RuntimeType
-import software.amazon.smithy.rust.codegen.core.smithy.generators.builderSymbol
-import software.amazon.smithy.rust.codegen.core.smithy.protocols.Protocol
-import software.amazon.smithy.rust.codegen.core.smithy.protocols.parse.EventStreamUnmarshallerGenerator
+import software.amazon.smithy.rust.codegen.client.testutil.clientIntegrationTest
 import software.amazon.smithy.rust.codegen.core.testutil.EventStreamTestModels
-import software.amazon.smithy.rust.codegen.core.testutil.EventStreamTestTools
-import software.amazon.smithy.rust.codegen.core.testutil.EventStreamTestVariety
-import software.amazon.smithy.rust.codegen.core.testutil.TestEventStreamProject
+import software.amazon.smithy.rust.codegen.core.testutil.EventStreamUnmarshallTestCases.writeUnmarshallTestCases
+import software.amazon.smithy.rust.codegen.core.testutil.IntegrationTestParams
+import software.amazon.smithy.rust.codegen.core.testutil.integrationTest
+import software.amazon.smithy.rust.codegen.core.testutil.unitTest
 
 class ClientEventStreamUnmarshallerGeneratorTest {
     @ParameterizedTest
     @ArgumentsSource(TestCasesProvider::class)
-    fun test(testCase: EventStreamTestModels.TestCase) {
-        EventStreamTestTools.runTestCase(
-            testCase,
-            object : ClientEventStreamBaseRequirements() {
-                override fun renderGenerator(
-                    codegenContext: ClientCodegenContext,
-                    project: TestEventStreamProject,
-                    protocol: Protocol,
-                ): RuntimeType {
-                    fun builderSymbol(shape: StructureShape): Symbol = shape.builderSymbol(codegenContext.symbolProvider)
-                    return EventStreamUnmarshallerGenerator(
-                        protocol,
-                        codegenContext,
-                        project.operationShape,
-                        project.streamShape,
-                        ::builderSymbol,
-                    ).render()
-                }
-            },
-            CodegenTarget.CLIENT,
-            EventStreamTestVariety.Unmarshall,
-        )
+    fun `test event stream unmarshaller generator`(testCase: EventStreamTestModels.TestCase) {
+        clientIntegrationTest(
+            testCase.model,
+            IntegrationTestParams(service = "test#TestService", addModuleToEventStreamAllowList = true),
+        ) { codegenContext, rustCrate ->
+            val crateName = codegenContext.moduleUseName()
+            val generator = "$crateName::event_stream_serde::TestStreamUnmarshaller"
```

Comment:
This is probably not good

Reply:
This might break with @drganjoo 's PR, #2256

Reply:
It didn't 🤔